### PR TITLE
[Info] add serialization

### PR DESCRIPTION
**Pull request by philbucher**
the implementation is quite basic for now, but I think it serves our purpose perfectly

### CODE REVIEW

File: co_sim_io/impl/info.hpp [hunk 1]
```diff
@@ -132,6 +138,52 @@ class Info
...
+            {"InfoData_string" , std::make_shared<Internals::InfoData<std::string>>("")}
+        };
+
+        std::string key, registered_name;
+
+        int size;
+        in >> size;
+
+        for (int i=0; i<size; ++i) {
+            in.ignore(10, '\n'); // skipping to end of line
```
Comment (pooyan-dadvand) [inline co_sim_io/impl/info.hpp:175]: What are you ignoring here? And why max to 10?

**Comment by philbucher** [comment]
What exactly do you mean with Tags? We used this name for different things already

File: co_sim_io/impl/info.hpp (hunk 1 above)
Comment (philbucher) [reply co_sim_io/impl/info.hpp:175]: Hm you spotted the good one :)
So I have one word per line
I read it with >>, but this doesn’t go to the next line, which I read with getline
Probably now makes sense to use >> there as well
I will try

Comment (philbucher) [reply co_sim_io/impl/info.hpp]: now it works without this

**Comment by pooyan-dadvand** [comment]
You are using the field name as the tag which is ok in your case and your test ouput is:
```
5
keyword
InfoData_string
awesome
is_converged
InfoData_bool
1
```
With tag it would be:
```
Size: 5
Key: keyword
Type: InfoData_string
Data: awesome
Key: is_converged
Type: InfoData_bool
Data: 1
```
Note that I usually write the tag only in debugging or it would be useful for eventually writing it in another format like json.

**Comment by philbucher** [comment]
> You are using the field name as the tag which is ok in your case and your test ouput is:
> 
> ```
> 5
> keyword
> InfoData_string
> awesome
> is_converged
> InfoData_bool
> 1
> ```
> 
> With tag it would be:
> 
> ```
> Size: 5
> Key: keyword
> Type: InfoData_string
> Data: awesome
> Key: is_converged
> Type: InfoData_bool
> Data: 1
> ```
> 
> Note that I usually write the tag only in debugging or it would be useful for eventually writing it in another format like json.

I see
I think I will leave it as is for now to not spend too much time on it, since most likely it will be extended/refactored in the future

thanks for the review

**Pull request merged**